packages/realm/src/errors.ts

```diff
@@ -103,11 +103,30 @@ export function fromBindingSyncError(error: binding.SyncError) {
   }
 }
 
+/**
+ * An class describing a sync error.
+ */
 export class SyncError extends Error {
   public name = "SyncError";
+
+  /**
+   * A numeric code representing the error.
+   */
   public code: number;
+
+  /**
+   * A string representing the error category.
+   */
   public category: string;
-  public isFatal: boolean;
```

Comment:
Removing `isFatal` would be a breaking change though right?

Reply:
You're correct, I saw it wrong, I thought we added it in the new version

Reply:
Yeah. If we want to do this, it could be deprecated here and the removal go into v13.

Reply:
Oh. I saw now that you just moved it down in the file 👍

Reply:
Yep, I added the deprecated tag now